packages/shared/lib/services/sync/config/config.service.ts

```diff
@@ -384,12 +378,12 @@ export async function deleteSyncConfig(id: number): Promise<void> {
     });
 }
 
-export async function disableScriptConfig(id: number): Promise<void> {
-    await schema().from<SyncConfig>(TABLE).where({ id }).update({ enabled: false });
+export async function disableScriptConfig({ id, environmentId }: { id: number; environmentId: number }): Promise<number> {
+    return await db.knex.from<SyncConfig>(TABLE).where({ id, environment_id: environmentId }).update({ enabled: false });
 }
 
-export async function enableScriptConfig(id: number): Promise<void> {
-    await schema().from<SyncConfig>(TABLE).where({ id }).update({ enabled: true });
+export async function enableScriptConfig({ id, environmentId }: { id: number; environmentId: number }): Promise<number> {
+    return await db.knex.from<SyncConfig>(TABLE).where({ id, environment_id: environmentId }).update({ enabled: true });
```

Comment:
isn't id not unique?

Reply:
id is unique but it's an auto-increment that is easily guessable, until we have a better permissions check it's safer to always query with environmentId for security reason